py2many/inference.py

```diff
@@ -105,6 +105,50 @@ def is_compatible(
     return True
 
 
+class LanguageInferenceBase:
+    """Base class for language-specific type inference"""
+
+    TYPE_MAP = {}  # Should be overridden by subclasses
+    CONTAINER_TYPE_MAP = {}  # Should be overridden by subclasses
+    WIDTH_RANK = {}  # Should be overridden by subclasses
+
+    @classmethod
+    def map_type(cls, typename):
+        typeclass = class_for_typename(typename, None)
+        if typeclass in cls.TYPE_MAP:
+            return cls.TYPE_MAP[typeclass]
+        return typename
```

Comment:
```suggestion
        return cls.TYPE_MAP.get(typeclass, typename)
```
% [`ruff rule SIM401`](https://docs.astral.sh/ruff/rules/if-else-block-instead-of-dict-get)